modules/402-ingress-nginx/images/controller-1-10/patches/new-metrics.patch

```diff
@@ -0,0 +1,171 @@
+diff --git a/internal/ingress/controller/store/store.go b/internal/ingress/controller/store/store.go
+index 284f53209..3e00ae2b5 100644
+--- a/internal/ingress/controller/store/store.go
++++ b/internal/ingress/controller/store/store.go
+@@ -55,6 +55,7 @@ import (
+ 	ngx_template "k8s.io/ingress-nginx/internal/ingress/controller/template"
+ 	"k8s.io/ingress-nginx/internal/ingress/defaults"
+ 	"k8s.io/ingress-nginx/internal/ingress/errors"
++	"k8s.io/ingress-nginx/internal/ingress/metric/collectors"
+ 	"k8s.io/ingress-nginx/internal/ingress/resolver"
+ 	"k8s.io/ingress-nginx/internal/k8s"
+ 	"k8s.io/ingress-nginx/pkg/apis/ingress"
+@@ -269,6 +270,8 @@ func New(
+ 		defaultSSLCertificate: defaultSSLCertificate,
+ 	}
+ 
++	collectors.WorkerStatusInstance.Set(store.backendConfig.MaxWorkerConnections, store.backendConfig.MaxWorkerOpenFiles, store.backendConfig.WorkerProcesses)
++
+ 	eventBroadcaster := record.NewBroadcaster()
+ 	eventBroadcaster.StartLogging(klog.Infof)
+ 	if !disableSyncEvents {
+@@ -1215,6 +1218,8 @@ func (s *k8sStore) setConfig(cmap *corev1.ConfigMap) {
+ 		s.backendConfig.UseGeoIP2 = false
+ 	}
+ 
++	collectors.WorkerStatusInstance.Set(s.backendConfig.MaxWorkerConnections, s.backendConfig.MaxWorkerOpenFiles, s.backendConfig.WorkerProcesses)
++
+ 	s.writeSSLSessionTicketKey(cmap, "/etc/ingress-controller/tickets.key")
+ }
+ 
+diff --git a/internal/ingress/metric/collectors/nginx_status.go b/internal/ingress/metric/collectors/nginx_status.go
+index f3afdc334..c9fe50594 100644
+--- a/internal/ingress/metric/collectors/nginx_status.go
++++ b/internal/ingress/metric/collectors/nginx_status.go
+@@ -17,13 +17,15 @@ limitations under the License.
+ package collectors
+ 
+ import (
+-	"log"
+-	"regexp"
+-	"strconv"
+-
+ 	"github.com/prometheus/client_golang/prometheus"
+ 	"k8s.io/ingress-nginx/internal/nginx"
+ 	"k8s.io/klog/v2"
++	"log"
++	"regexp"
++	"runtime"
++	"strconv"
++	"sync"
++	"syscall"
+ )
+ 
+ var (
+@@ -34,6 +36,53 @@ var (
+ 	waiting = regexp.MustCompile(`Waiting: (\d+)`)
+ )
+ 
++type WorkerStatus struct {
++	WorkerConnections  int
++	WorkerProcesses    int
++	WorkerRlimitNofile int
++	Lock               sync.Mutex
++}
```

Comment:
should've been defined as a private type and fields :)

Reply:
Should, but it is just a patch, next time I will make them private :)